test_improvements.py

```
@@ -0,0 +1,491 @@
+    """Test VLM engine integration improvements."""
+    
+    def test_vlm_engines_receive_images_not_paths(self):
+        """Test that VLM engines receive PIL Image objects, not file paths."""
+        # Create mock image
+        test_image = Image.new('RGB', (100, 100), color='white')
+        
+        with patch('pipeline._mistral_ocr') as mock_mistral:
+            mock_mistral.return_value = pipeline.OcrResult("test", ["test"], [0.9])
+            
+            # Test mistral engine receives image object
+            result = pipeline._run_single_engine_with_cache(
```

Flake8 (7.2.0)

[error] local variable 'result' is assigned to but never used

(F841)